quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocomplete-adapter-utills.ts

```diff
@@ -0,0 +1,164 @@
+import { findLastDotIndex } from "../../runner/services/autocomplete/highlight-and-score";
+import { TableInfo } from "../sql-context-evaluator";
+import { trinoToJs } from "./trinoToJs"
+
+interface ObjectChild {
+  name: string;
+  dataType: any;
+}
+
+export function getObjectChildren(obj: Record<string, any>, parentName = ''): ObjectChild[] {
+  const children: ObjectChild[] = [];
+
+  for (const key in obj) {
+    const childName = parentName ? `${parentName}.${key}` : key;
+    if (typeof obj[key] === 'object' && obj[key] !== null) {
+      children.push({ name: childName, dataType: obj[key] });
+      children.push(...getObjectChildren(obj[key], childName));
+    } else {
+      children.push({ name: childName, dataType: obj[key] });
+    }
+  }
+  return children;
+}
+
+export function findRelevantPartOfPrefix(tables: TableInfo[], brokenPrefix: string[]): string {
+  // Remove empty cells from brokenPrefix array
+  brokenPrefix = brokenPrefix.filter(cell => cell.trim() !== '');
```

Comment:
cell => item

Reply:
Eliminate Side Effect of changing broken Prefix.

Reply:
Eliminated  Side Effect by changing when called and not in function itself